frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx

```diff
@@ -0,0 +1,169 @@
+import { kmsConfigValidation } from './utils';
+import { KMSConfigureProps } from './providers';
+import { advancedHpcsModal } from '../modals/advanced-kms-modal/advanced-ibm-kms-modal';
+import { HpcsConfig, ProviderNames } from '../../types';
+import './kms-config.scss';
+
+export const HpcsConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, className }) => {
+  const { t } = useTranslation();
+
+  const kms: HpcsConfig = state.kms?.[ProviderNames.HPCS];
+  const kmsObj: HpcsConfig = _.cloneDeep(kms);
+
+  React.useEffect(() => {
+    const hasHandled: boolean = kmsConfigValidation(kms, ProviderNames.HPCS);
+    if (kms.hasHandled !== hasHandled)
+      dispatch({
+        type: 'securityAndNetwork/setHpcs',
+        payload: {
+          ...kms,
+          hasHandled,
+        },
+      });
+  }, [dispatch, kms]);
+
+  const setServiceName = (name: string) => {
+    kmsObj.name.value = name;
+    kmsObj.name.valid = name !== '';
+    dispatch({
+      type: 'securityAndNetwork/setHpcs',
+      payload: kmsObj,
+    });
+  };
+
+  const setInstanceId = (instanceId: string) => {
+    kmsObj.instanceId.value = instanceId;
+    kmsObj.instanceId.valid = instanceId !== '';
+    dispatch({
+      type: 'securityAndNetwork/setHpcs',
+      payload: kmsObj,
+    });
+  };
+
+  const setApiKey = (apiKey: string) => {
+    kmsObj.apiKey.value = apiKey;
+    kmsObj.apiKey.valid = apiKey !== '';
+    dispatch({
+      type: 'securityAndNetwork/setHpcs',
+      payload: kmsObj,
+    });
+  };
+
+  const setRootKey = (rootKey: string) => {
+    kmsObj.rootKey.value = rootKey;
+    kmsObj.rootKey.valid = rootKey !== '';
+    dispatch({
+      type: 'securityAndNetwork/setHpcs',
+      payload: kmsObj,
+    });
+  };
+
+  const openAdvancedModal = () =>
+    advancedHpcsModal({
+      state,
+      dispatch,
+    });
+
+  const isValid = (value: boolean) => (value ? ValidatedOptions.default : ValidatedOptions.error);
+
+  return (
+    <>
+      <FormGroup
+        fieldId="kms-service-name"
+        label={t('ceph-storage-plugin~Service name')}
+        className={`${className}__form-body`}
+        helperTextInvalid={t('ceph-storage-plugin~This is a required field')}
+        validated={isValid(kms.name?.valid)}
+        helperText={t(
+          'ceph-storage-plugin~A unique name for the key management service within the project.',
+        )}
+        isRequired
+      >
+        <TextInput
+          value={kms.name?.value}
+          onChange={setServiceName}
+          type="text"
+          id="kms-service-name"
+          name="kms-service-name"
+          isRequired
+          validated={isValid(kms.name?.valid)}
+          data-test="kms-service-name-text"
+        />
+      </FormGroup>
+      <FormGroup
+        fieldId="kms-instance-id"
+        label={t('ceph-storage-plugin~Service instance ID')}
+        className={`${className}__form-body`}
+        helperTextInvalid={t('ceph-storage-plugin~This is a required field')}
```

Comment:
lets extract it outside and then use it everywhere.

Reply:
even if I extract it out, it is going to be the same, for each field I have to maintain different param, it won't make any difference. 

If i create a generic function will affect the flexible way of changing the param in future